torch/distributions/constraints.py

```diff
@@ -697,41 +782,72 @@ def event_dim(self) -> int:  # type: ignore[override]
             dim += 1
         return dim
 
-    def check(self, value):
+    def check(self, value: Tensor) -> Tensor:
         assert -value.dim() <= self.dim < value.dim()
         vs = [value.select(self.dim, i) for i in range(value.size(self.dim))]
         return torch.stack(
             [constr.check(v) for v, constr in zip(vs, self.cseq)], self.dim
         )
 
 
+# Type aliases.
+Dependent: TypeAlias = _Dependent
```

Comment:
Why are we exporting the TypeAlias (you can suppress the ruff PYI rule for backwards compatibility if that is the only reason FYI).

Reply:
Because of the `test_public_bindings`check: https://github.com/pytorch/pytorch/actions/runs/15327668392/job/43132535150

Reply:
I am a bit confused, values prefixed by _ shouldn't be exported by default?

Reply:
Alternatively, maybe one could clean this whole thing up a bit as follows:

1. Make the classes `_Interval`, `_independent`, etc. themselves public
2. Create one class per constraint of interest (for instance one could make a `UnitInterval` subtype of `Interval`
3. For classes that do not require parameters, create one canonical instance and export it as well

Reply:
Values prefixed by `_` are considered private by convention in most of the python ecosystem. `__all__` is supposed to contain the public interface of the module.

Reply:
Why are these classes public though? There is also overloading here, there is public to users (which we very conservative about, and then there is public for internal APIs.

Reply:
I see, just because the singletons are public, we do need the TypeAlias to be?

Reply:
Technically, you could use `type[constraints.unit_interval]`, but I see your point?